core/src/main/java/org/elasticsearch/plugins/PluginInfo.java

```diff
@@ -95,6 +95,12 @@ public static PluginInfo readFromProperties(Path dir) throws IOException {
         if (version == null) {
             throw new IllegalArgumentException("Property [version] is missing for plugin [" + name + "]");
         }
+        String pluginName = props.getProperty("name");
```

Comment:
Now the logging here is really "fun". Maybe pull the pluginName to the top and use it in the []s. I'm trying to think through what this will look like for the temporary directory versions. Maybe it makes sense to make the whole temporary directory thing a more first class citizen - like a field or a subclass of this rather than just passing in randomness.